validator/src/main.rs

```diff
@@ -1051,33 +1037,6 @@ pub fn main() {
                 _ => panic!("Unsupported ledger_compression: {ledger_compression_string}"),
             },
         },
-        shred_storage_type: match matches.value_of("rocksdb_shred_compaction") {
```

Comment:
also remove `rocksdb_shred_compaction` from validator/src/cli.rs?

Reply:
Two relevant args:
- `--rocksdb-shred-compaction`: I left this one so that people who specify `fifo` get an error about illegal value instead of unknown flag. Also, the need for this arg probably goes away in the long term, but we'll be spending some time on rocksdb in the next couple weeks so figured I'd leave it for a little bit incase we make any changes
- `rocksdb-fifo-shred-storage-size`: This arg is specifically for `fifo`, I will be marking it deprecated in a separate PR

Reply:
for `--rocksdb-shred-compaction` since we removed the argument parsing users who pass `fifo` would not get an error right? it would just silently use `level` instead?

If you want to keep the arg wdyt about:
- Updating the help message to no longer display "fifo"
- check `matches.value_of("rocksdb_shred_compaction")` here just to panic if it's not empty or `level`

And then in the follow up PR we can mark `--rocksdb-fifo-shred-storage-size` as deprecated as you suggested so that it complains when people try to set it.

Reply:
> for --rocksdb-shred-compaction since we removed the argument parsing users who pass fifo would not get an error right? it would just silently use level instead?

No, it'll error. In https://github.com/anza-xyz/agave/pull/3451, I removed `fifo` as an option from `.possible_values()`. CLAP will yell at a user for passing a value that is not in the array passed to `.possible_values()`

Reply:
ah forgot you already did that haha
don't mind me 🙃

Reply:
For completeness, here is the error:
```
exec /home/sol/src/agave/target/release/agave-validator ... --rocksdb-shred-compaction fifo ...

error: 'fifo' isn't a valid value for '--rocksdb-shred-compaction <ROCKSDB_COMPACTION_STYLE>'
	[possible values: level]
```